src/escenas/libroPrimaria/EscenaChuy.ts

```diff
@@ -0,0 +1,113 @@
+/// <reference path = "EscenaDesdeMapa.ts" />
+
+class EscenaChuy extends EscenaDesdeMapa {
+	automata: Chuy;
+	xFinal: number;
+	yFinal: number;
+
+	static clasesDeActoresInvolucrados(): typeof ActorAnimado[] {
+		return [Chuy, Trofeo, Paleta, PelotaAnimada, Pulpito, PingPong];
+	};
+
+	static pathFondo(): string {
+		return 'fondo.chuy.png';
+	}
+
+	static imagenesAdicionales(): string[] {
+		return Casilla.imagenesPara('chuy').concat(Obstaculo.imagenesPara('chuy'));
+	}
+
+	constructor(especificacion: Spec, opciones?: opcionesMapaAleatorio, posFinal?: [number, number]) {
+		super();
+		this.initDesdeUnaOVariasDescripciones(especificacion, opciones);
+
+		if (posFinal) {
+			this.xFinal = posFinal[0];
+			this.yFinal = posFinal[1];
+		}
+	}
+
+	ajustarGraficos() {
+		this.automata.escala *= this.escalaSegunCuadricula(2.5);
+		this.automata.setY(this.automata.getY() + this.automata.getAlto() / 4);
+		this.automata.setX(this.automata.getX() + 15)
+
+		this.obtenerActoresConEtiquetas(["Trofeo", "Paleta", "Pulpito"]).forEach(actor => {
+			actor.aprender(Flotar, { Desvio: 4 });
+			actor.escala *= this.escalaSegunCuadricula(0.5);
+		});
+
+		this.obtenerActoresConEtiqueta("PelotaAnimada").forEach(actor => {
+			actor.aprender(Flotar, { Desvio: 4 });
+			actor.escala *= this.escalaSegunCuadricula(1) * 0.25;
+		});
+
+		this.obtenerActoresConEtiqueta("PingPong").forEach(actor => {
+			actor.aprender(Flotar, { Desvio: 4 });
+			actor.escala *= this.escalaSegunCuadricula(1) * 0.15;
+		});
+
+
+		this.obtenerActoresConEtiqueta("Obstaculo").forEach(actor => {
+			actor.escala *= this.escalaSegunCuadricula(0.7);
+		});
+
+	}
+
+	mapearIdentificadorAActor(id, nroFila, nroColumna): ActorAnimado {
+		switch (id) {
+			case 'A': return this.automata;
+			case 'O': return this.obtenerObstaculo(nroFila, nroColumna);
+			case 'T': return new Trofeo();
+			case 'E': return new Paleta();
+			case 'U': return new Pulpito();
+			case 'P': return new PingPong();
+			case 'G': return new PelotaAnimada(0,0);
+			default: throw new Error("El identificador '" + id +
+				"' no es válido en una escena de Chuy.");
+		}
+	}
+
+	obtenerAutomata(): Chuy {
+		return new Chuy();
+	}
+
+	obtenerObstaculo(fila: number, columna: number): Obstaculo {
+		let archivosObstaculos = ["obstaculo.chuy1.png", "obstaculo.chuy2.png", "obstaculo.chuy3.png", "obstaculo.chuy4.png"];
+		return new Obstaculo(archivosObstaculos, (fila + 1) + (fila + 1) * (columna + 1));
+	}
+
+	estaEnPosicionFinalSiLaTiene(): boolean {
+		return !this.xFinal || this.automata.casillaActual().sos(this.yFinal, this.xFinal) || this.automata.alFinalDelCamino();
+	}
```

Comment:
La lógica de esto es: 

- si no necesita estar en una posición final no la va a tener
- si la tiene me fijo que esté en esa posición
- si quiero saber si la posición es que haya llegado al final del camino, le pongo una posición al generar el mapa que nunca pueda existir (como [-1,0]) para que llegue a la última condición.

Reply:
Cuidado, si xFinal es 0 ese not da verdadero. Compará con triple igual. 

```suggestion
	estaEnPosicionFinalSiLaTiene(): boolean {
		return this.xFinal === undefined || this.automata.casillaActual().sos(this.yFinal, this.xFinal) || this.automata.alFinalDelCamino();
	}
```